### PR TITLE
docs: add a method to deploy vuepress site with github actions

### PR DESCRIPTION
**Pull request by jenkey2011**
**Summary**

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [x] Docs
- [ ] Build-related changes
- [ ] Other, please describe:

If changing the UI of default theme, please provide the **before/after** screenshot:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [ ] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)

You have tested in the following browsers: (Providing a detailed version will be better.)

- [ ] Chrome
- [ ] Firefox
- [ ] Safari
- [ ] Edge
- [ ] IE

If adding a **new feature**, the PR's description includes:

- [ ] A convincing reason for adding this feature
- [ ] Related documents have been updated
- [ ] Related tests have been updated

To avoid wasting your time, it's best to open a **feature request issue** first and wait for approval before working on it.

**Other information:**

### CODE REVIEW

File: packages/docs/docs/guide/deploy.md [hunk 1]
```diff
@@ -56,6 +56,35 @@ cd -
 You can also run the above script in your CI setup to enable automatic deployment on each push.
 :::
 
+### GitHub Pages and GitHub Actions
+
+1. Create a personal access token
+2. Create encrypted secrets under your repository
+3. Create a `.yml` or `.yaml` file in the `.github/workflows` directory in the root of your repository. E.g:`vueoress-deploy.yml`;
```
Comment (kefranabg) [inline packages/docs/docs/guide/deploy.md]:
```suggestion
3. Create a `.yml` or `.yaml` file in the `.github/workflows` directory in the root of your repository. E.g:`vuepress-deploy.yml`;
```

Comment (jenkey2011) [reply packages/docs/docs/guide/deploy.md]: Thank you for your careful review!! I recommited a version  according to the suggestion.
About the minimalist repos for testing, please see the comment below

**Comment by jenkey2011** [comment]
**repository**: [https://github.com/jenkey2011/vuepress-deploy-demo](https://github.com/jenkey2011/vuepress-deploy-demo)

**detail:**

The following guides are based on some shared assumptions:

- Source code repository(`vuepress-deploy-demo`), we call it `A`
- Target repository(`vuepress-deploy-demo-target`), we call it `B`

There are three situations. Each situation corresponds to a deployment file. 

- `A` TO `A:gh_pages`: `deploy-to-current-repo-gh_pages.yml`
- `A` TO `B:master`: `deploy-to-other-repo-master.yml`
- `A` TO `B:custom_branch`: `deploy-to-other-repo-custom_branch.yml`

## Here we can see

- The action: [https://github.com/jenkey2011/vuepress-deploy-demo/actions](https://github.com/jenkey2011/vuepress-deploy-demo/actions)

- `A` TO `A:gh_pages`: [https://github.com/jenkey2011/vuepress-deploy-demo/tree/gh_pages](https://github.com/jenkey2011/vuepress-deploy-demo/tree/gh_pages)

- `A` TO `B:master`:[https://github.com/jenkey2011/vuepress-deploy-demo-target/tree/master](https://github.com/jenkey2011/vuepress-deploy-demo-target/tree/master)

- `A` TO `B:custom_branch`:[https://github.com/jenkey2011/vuepress-deploy-demo-target/tree/custom_branch](https://github.com/jenkey2011/vuepress-deploy-demo-target/tree/custom_branch)

- The GitHub Pages of the repository B： [https://jenkey2011.github.io/vuepress-deploy-demo-target/](https://jenkey2011.github.io/vuepress-deploy-demo-target/)

**Comment by jenkey2011** [comment]
Hi @kefranabg , hope you can take a look 😆😆😆

**Comment by kefranabg** [comment]
Sorry, so much things to do 😅 I'll try to review ASAP !

**Review by jenkey2011 — COMMENTED** [review]
Any one?

**Review by ulivz — APPROVED** [review]

**Pull request merged**